fhttp/http_client.go

```diff
@@ -145,6 +147,24 @@ func (h *HTTPOptions) ResetHeaders() {
 func (h *HTTPOptions) InitHeaders() {
 	h.ResetHeaders()
 	h.extraHeaders.Add("User-Agent", userAgent)
+	err := h.ValidateAndAddBasicAuthentication()
+	if err != nil {
+		log.Debugf("User credential is not valid. %v", err)
+	}
+}
+
+// ValidateAndAddBasicAuthentication validates user credentials and adds basic authentication to http header,
+// if user credentials are valid.
+func (h *HTTPOptions) ValidateAndAddBasicAuthentication() error {
+	if len(h.UserCredentials) <= 0 {
+		return nil // user credential is not entered
+	}
+	s := strings.SplitN(h.UserCredentials, ":", 2)
+	if len(s) != 2 {
+		return fmt.Errorf("invalid user credentials are used %s. Expected format user:password", h.UserCredentials)
```

Comment:
or log the error here

but right now it's not logged if I pass -user foo is it ? please paste the full output of
```
fortio load -user foo www.google.com
```

Reply:
Fortio 1.0.0-pre running at 8 queries per second, 4->4 procs, for 5s: www.google.com
07:53:18 I httprunner.go:82> Starting http test for www.google.com with 4 threads at 8.0 qps
07:53:18 E http_client.go:152> User credential is not valid. invalid user credentials are used foo. Expected format user:password
07:53:18 W http_client.go:105> assuming http:// on missing scheme for 'www.google.com'
Starting at 8 qps with 4 thread(s) [gomax 4] for 5s : 10 calls each (total 40)
07:53:24 I periodic.go:533> T003 ended after 5.102056123s : 10 calls. qps=1.9599941198059614
07:53:24 I periodic.go:533> T002 ended after 5.10392474s : 10 calls. qps=1.9592765390188727
07:53:24 I periodic.go:533> T001 ended after 5.104550597s : 10 calls. qps=1.959036316708685
07:53:24 I periodic.go:533> T000 ended after 5.109756345s : 10 calls. qps=1.957040478023028
Ended after 5.109886411s : 40 calls. qps=7.828
Sleep times : count 36 avg 0.44596145 +/- 0.009171 min 0.413167981 max 0.45571464 sum 16.0546124
Aggregated Function Time : count 40 avg 0.10895143 +/- 0.008826 min 0.099699288 max 0.142015374 sum 4.35805737
# range, mid point, percentile, count
>= 0.0996993 <= 0.1 , 0.0998496 , 2.50, 1
> 0.1 <= 0.12 , 0.11 , 85.00, 33
> 0.12 <= 0.14 , 0.13 , 97.50, 5
> 0.14 <= 0.142015 , 0.141008 , 100.00, 1
# target 50% 0.111515
# target 75% 0.117576
# target 90% 0.128
# target 99% 0.141209
# target 99.9% 0.141935
Sockets used: 4 (for perfect keepalive, would be 4)
Code 200 : 40 (100.0 %)
Response Header Sizes : count 40 avg 693 +/- 0 min 693 max 693 sum 27720
Response Body/Total Sizes : count 40 avg 11884.25 +/- 18.59 min 11853 max 11926 sum 475370
All done 40 calls (plus 4 warmup) 108.951 ms avg, 7.8 qps

Reply:
In this output your above comment, which is changing log to err or warring from debug,  is applied.

Reply:
pro tip bracket command output with 3 backticks : ```
to get a pre formattted block like
```
this 
is 
an example
```

Reply:
``` test ```